CTP/shell/src/com/navercorp/cubridqa/shell/deploy/DeployOneNode.java

```diff
@@ -219,6 +229,92 @@ private void deploy_ctp() {
 		}
 	}
 	
+	private void updateCUBRIDConfigurations(){
+		String cubridPortId, brokerFirstPort, brokerSecondPort, cubridHAPortId;
+		
+		cubridPortId = context.getProperty("env." + this.currentEnvId + ".cubrid.cubrid_port_id");
+		brokerFirstPort = context.getProperty("env." + this.currentEnvId + ".broker1.BROKER_PORT");
+		brokerSecondPort = context.getProperty("env." + this.currentEnvId + ".broker2.BROKER_PORT");
+		cubridHAPortId = context.getProperty("env." + this.currentEnvId + ".ha.ha_port_id");
+		
+		cubridPortId = cubridPortId == null ?  context.getProperty("default.cubrid.cubrid_port_id") : cubridPortId;
+		brokerFirstPort = brokerFirstPort == null ?  context.getProperty("default.broker1.BROKER_PORT") : brokerFirstPort;
+		brokerSecondPort = brokerSecondPort == null ?  context.getProperty("default.broker2.BROKER_PORT") : brokerSecondPort;
+		cubridHAPortId = cubridHAPortId == null ?  context.getProperty("default.ha.ha_port_id") : cubridHAPortId;
+		
+		if (cubridPortId == null && brokerFirstPort == null
+				&& brokerSecondPort == null && cubridHAPortId == null)
+			return;
+		
+		ShellInput scripts = new ShellInput();
+		if(cubridPortId!=null){
+			scripts.addCommand("ini.sh -s common -u cubrid_port_id=" + cubridPortId + " $CUBRID/conf/cubrid.conf");
+			scripts.addCommand("ini.sh -s 'broker' -u MASTER_SHM_ID=" + cubridPortId + " $CUBRID/conf/cubrid_broker.conf");
+		}
+		
+		if(brokerFirstPort!=null){
+			scripts.addCommand("ini.sh -s '%query_editor' -u BROKER_PORT=" + brokerFirstPort + " $CUBRID/conf/cubrid_broker.conf");
+			scripts.addCommand("ini.sh -s '%query_editor' -u APPL_SERVER_SHM_ID=" + brokerFirstPort + " $CUBRID/conf/cubrid_broker.conf");
+		}
+		
+		if(brokerSecondPort!=null){
+			scripts.addCommand("ini.sh -s '%BROKER1' -u BROKER_PORT=" + brokerSecondPort + " $CUBRID/conf/cubrid_broker.conf");
+			scripts.addCommand("ini.sh -s '%BROKER1' -u APPL_SERVER_SHM_ID=" + brokerSecondPort + " $CUBRID/conf/cubrid_broker.conf");
+		}
+		
+		if(cubridHAPortId!=null && context.isHAMode()){
```

Comment:
It's better not to set port. This is because all ha parameters have been disabled as default.

$CUBRID/conf/cubrid_ha.conf
```
#[common]
#ha_port_id=59901
#ha_node_list=foo@apple:banana
#ha_db_list=tdb
#ha_apply_max_mem_size=300
#ha_copy_sync_mode=sync:sync
#ha_copy_log_max_archives=1
```